src/engraving/dom/edit.cpp

```diff
@@ -1456,6 +1467,12 @@ void Score::cmdAddTimeSig(Measure* fm, staff_idx_t staffIdx, TimeSig* ts, bool l
                     masterTimeSigs[nsig->track()] = nsig;
                 }
             }
+
+            if (local) {
+                select(ots, SelectType::SINGLE, staffIdx);
+            } else {
+                selectElementsWithSameTypeOnSegment(ElementType::TIMESIG, seg);
+            }
```

Comment:
Here, `seg` _is_ correctly set to a segment of the `score` score. But in this case too, I think it would make more sense to select the timesig only in the current score, i.e. `if (score == this) { … }`.